docs/blog/2019-03-25-behind-the-scenes-what-makes-gatsby-great/index.md

```diff
@@ -0,0 +1,496 @@
+
+**Data-constrained devices**
+
+One of the most appealing things about using a framework like Gatsby is that we can bake-in micro-optimizations and make these available in incremental updates to all of _our_ users making the performance optimizations available to all of _your_ users. In particular, later versions of Chrome support an API that grants network information like type of connection (e.g. `slow-2g`, `2g`, etc.), whether the browser has a `saveData` hint enabled, and more. With the help of [@addyosmani](https://twitter.com/addyosmani) we've baked this into our preloading logic. On devices with these hints or on slow connections, we disable the preloading to save data! It looks a little something like:
+
+```js
+// Skip prefetching if we know user is on slow or constrained connection
+if (`connection` in navigator) {
+  if ((navigator.connection.effectiveType || ``).includes(`2g`)) {
+    return false
+  }
+  if (navigator.connection.saveData) {
+    return false
+  }
+}
+```
+
+See the [code implementing this functionality here](https://github.com/gatsbyjs/gatsby/blob/be023447eb0f5bd4612a447098fc85c7afdf0cdf/packages/gatsby/cache-dir/loader.js#L227-L235).
+
+### `srcset` powering Responsive Images
+
+Whew. It's been quite a ride. Finally, I'd like to discuss one of my favorite Gatsby components, specifically [`gatsby-image`](/packages/gatsby-image/). This component _also_ registers an `IntersectionObserver`, and uses it in an interesting, performance-focused manner. Let's get to it.
+
+#### `gatsby-image`
+
+There are far, far, far too many otherwise decently performing websites that load a gargantuan 5Mb image request as a hero image. But--thankfully, this isn't you! You use `gatsby-image` and you get all of the following:
+
+- Responsive, optimized images using a `srcset`
+  - A [`picture`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture) element with a `source srcset` is used
+  - This means that using several media queries, you load the smallest image that matches your device (e.g. mobile devices get smaller images, desktop devices get larger images, etc.)
+  - We even generate 2x and 3x DPi images for beautiful images, regardless of the screen quality!
+- A base64 blurred image loaded by default
+  - This has two wins: 1) Larger images outside the viewport are not requested until they're needed, and 2) The blurred image is in a container with the same dimensions as the real image--therefore, no jumping when the image loads!
+  - Also see: [traced SVGs for a super slick alternative](/packages/gatsby-plugin-sharp/#tracedsvg)
+- An `IntersectionObserver` that swaps the base image for the larger image, when the image is in the viewport
+  - See [the code here](https://github.com/gatsbyjs/gatsby/blob/7684b4f0126724521967f7c8d31d32ec8d9b5fa6/packages/gatsby-image/src/index.js#L45-L69)
```

Comment:
```suggestion
  - See the [image IntersectionObserver code here](https://github.com/gatsbyjs/gatsby/blob/7684b4f0126724521967f7c8d31d32ec8d9b5fa6/packages/gatsby-image/src/index.js#L45-L69)
```
This might be a little long and/or repetitive...but we should write descriptive links whenever possible. Maybe there's something better?